src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.h

```diff
@@ -0,0 +1,290 @@
+
+    PushAvStreamTransportServer * mPushAvStreamTransportServer = nullptr;
+
+    /**
+     * This method is used by the SDK to set the PushAvStreamTransportServer pointer member in the delegate.
+     * This is done in the constructor during the instantiation of the PushAvStreamTransportServer object.
+     *
+     * @param aPushAvStreamTransportServer A pointer to the PushAvStreamTransportServer object related to this delegate object.
+     */
+    void SetPushAvStreamTransportServer(PushAvStreamTransportServer * aPushAvStreamTransportServer)
+    {
+        mPushAvStreamTransportServer = aPushAvStreamTransportServer;
+    }
+
+protected:
+    PushAvStreamTransportServer * GetPushAvStreamTransportServer() const { return mPushAvStreamTransportServer; }
+};
+
+class PushAvStreamTransportServer : private AttributeAccessInterface, private CommandHandlerInterface
+{
+public:
+    /**
+     * Creates a Push AV Stream Transport server instance. The Init() function needs to be called for this instance to be registered
+     * and called by the interaction model at the appropriate times.
+     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aDelegate A reference to the delegate to be used by this server.
+     * Note: the caller must ensure that the delegate lives throughout the instance's lifetime.
+     */
+    PushAvStreamTransportServer(EndpointId endpointId, PushAvStreamTransportDelegate & delegate, const BitFlags<Feature> aFeature,
+                                PersistentStorageDelegate & aPersistentStorage);
+
+    ~PushAvStreamTransportServer() override;
+
+    /**
+     * @brief Initialise the Push AV Stream Transport server instance.
+     * This function must be called after defining an PushAvStreamTransportServer class object.
+     * @return Returns an error if the given endpoint and cluster ID have not been enabled in zap or if the
+     * CommandHandler or AttributeHandler registration fails, else returns CHIP_NO_ERROR.
+     * This method also checks if the feature setting is valid, if invalid it will return CHIP_ERROR_INVALID_ARGUMENT.
+     */
+    CHIP_ERROR Init();
+
+    bool HasFeature(Feature feature) const;
+
+    // Attribute Getters
+    BitMask<SupportedContainerFormatsBitmap> GetSupportedContainerFormats() const { return mSupportedContainerFormats; }
+    BitMask<SupportedIngestMethodsBitmap> GetSupportedIngestMethods() const { return mSupportedIngestMethods; }
+
+    // Helper functions
+    bool FindStreamTransportConnection(const uint16_t connectionID);
+    uint16_t GenerateConnectionID();
+
+    // Add/Remove Management functions for transport
+    CHIP_ERROR AddStreamTransportConnection(const uint16_t transportConnectionId);
+
+    CHIP_ERROR RemoveStreamTransportConnection(const uint16_t transportConnectionId);
+
+private:
+    PushAvStreamTransportDelegate & mDelegate;
+    EndpointId mEndpointId;
+    const BitFlags<Feature> mFeature;
```

Comment:
Is the default storage provided by the attribute store insufficient? Why must we maintain these attributes within the cluster server itself?